File: TP file d'entier/Code Java/src/ThreadProducteur.java
import java.util.Random;

import javax.swing.JLabel;

public class ThreadProducteur extends Thread {
	private final File f;
	private final int timeSleep;
	private final JLabel labProd;
	private final JLabel labFile;
	private Random generateur = new Random(java.lang.System.currentTimeMillis()) ;
	private static final int NB_MAX = 100;
	private static final int NB_MIN = 1;
	
	
	public ThreadProducteur(File  f, int timeSleep,JLabel labProd,JLabel labFile, String nom) {
		super(nom);
		this.timeSleep=timeSleep;
		this.f=f;
		this.labFile=labFile;
		this.labProd=labProd;
	}
	
	public void run() {
			try {
				while(!interrupted()) {
				int e = generateur.nextInt(NB_MAX) + NB_MIN;
				labProd.setText(getName()+f.enfiler(e));
				labFile.setText(f.toString());
				labProd.repaint(); labFile.repaint();
				sleep(timeSleep);
				}
			} catch (InterruptedException e) {}
		}
}

File: TP file d'entier/Code Python/ThreadProducteur.py
# coding: utf-8

from threading import Thread
import time
import random
                    
#______________Class ThreadProducteur____________________________________________________


class ThreadProducteur(Thread):

    def __init__(self, f, tempsSommeil, labProd, labFile, nom):
        Thread.__init__(self)
        self.tempsSommeil=tempsSommeil/1000
        self.f=f
        self.labFile=labFile
        self.labProd=labProd
        self.name=nom
        self.interrupted = False



    #*******_Methode_******
        
    def interrupt(self):
        self.interrupted = True

    def run(self):
        while not self.interrupted:
            e = random.randint(1,100)
            
            self.labProd.config(text=self.name+self.f.enfiler(e))
            self.labFile.config(text=self.f.toString())
            
            self.labProd.update()
            self.labFile.update()
            time.sleep(self.tempsSommeil)
